### ISSUE TITLE
Debian pygame package should remove unused font dependencies (305)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#320]
#  **Issue &numero;305 opened by *[illume](https://github.com/illume)* at 2016-07-14 10:46:18**

Originally reported by: **PanderMusubi (Bitbucket: [PanderMusubi](https://bitbucket.org/PanderMusubi), GitHub: [PanderMusubi](https://github.com/PanderMusubi))**

----------------------------------------

Minimalise font dependencies in terms of:

* no font .deb package dependencies in the pygame .deb
* no usage of absolute font file paths
* use font config where possible
* when absolutely necessary shipping font files only in the source code for testing purposes, not in the final build
* when absolutely necessary refering to TTF files, also look for OTF files when those files are not present as graceful degradation
* et cetera

See also:

* https://bitbucket.org/pygame/pygame/issues/306/separate-build-and-install-package
* https://bitbucket.org/pygame/pygame/issues/107/pygame-192-release

----------------------------------------
- Bitbucket: https://bitbucket.org/pygame/pygame/issue/305
<hr>

#  Comments

# # #  **[illume](https://github.com/illume)* commented at 2016-08-05 18:02:07*

*Original comment by* **René Dudfield (Bitbucket: [illume](https://bitbucket.org/illume), GitHub: [illume](https://github.com/illume))**:

----------------------------------------

This needs to be followed up with the various .deb maintainers I guess.

Or is there anything inside pygame itself that needs fixing?

<hr>

# # #  **[illume](https://github.com/illume)* commented at 2016-08-06 09:42:44*

*Original comment by* **PanderMusubi (Bitbucket: [PanderMusubi](https://bitbucket.org/PanderMusubi), GitHub: [PanderMusubi](https://github.com/PanderMusubi))**:

----------------------------------------

Yes, .deb maintainers.

I did some inspection. If you look at the output of

```grep '\.[ot]tf' -riI *|grep -v docscomments.json```

you will see that the .deb packages do not need these dependencies. Absolute paths are not used and most of the more demanding font code are tests and examples. so if you agree, we don't need to change anything in pygame, only drop deps in .debs and this can be closed for 1.9.2 too.

<hr>

# # #  **[MyreMylar](https://github.com/MyreMylar)* commented at 2020-04-29 19:47:42*

I feel like the conclusion of this report is that nothing needs to change in the pygame project. There is one font in the codebase:  `freesansbold.ttf` which is the default pygame font.

How does one follow up with .deb maintainers? What would we tell them if we had a list of their email addresses?
<hr>

# # #  **[MyreMylar](https://github.com/MyreMylar)* commented at 2020-05-16 13:34:32*

**To do to close:**

- [ ] Verify that debian is no longer bundling unnecessary fonts with pygame package. Not sure exactly where to follow this up.
<hr>

# # #  **[nthykier](https://github.com/nthykier)* commented at 2020-09-08 18:34:33*

@MyreMylar: AFAICT, the Debian maintainers are "unvendoring" the font file provided by pygame upstream.  This is based on the following logic:

 * The Debian pygame package depend exactly on one font-package, `fonts-freefont-ttf`.
 * The build scripts for the Debian pygame package has scripts to replace the embedded `freesansbold.ttf` file with a file with a similar name (see https://salsa.debian.org/python-team/modules/pygame/-/blob/master/debian/rules# L5).
 * This pattern follows a tradition of Debian tracking and trying to remove duplicated font files: https://lintian.debian.org/tags/duplicate-font-file.html

In other words, I think this is "business as usual" and not a real problem.

(I think the original issue was about us using fontconfig to find a similar font instead of embedding one, which is a different problem than the Debian package providing our embedded font from a separate package)
<hr>

# # #  **[illume](https://github.com/illume)* commented at 2020-09-09 06:32:47*

Yeah, I'll close this.
